# Genomic Analysis and Phylodynamics Workshop

## Pre-course Instructions

Please following these instructions to download all programs and data for the course. You will require ~ 1GB of space to install all programs and data.

### Data 

Please download all data required for this course by clicking the following link (Note, the file size is ~ MB):

[Download Data](https://drive.google.com/drive/folders/1w3WG0jCj9BfB6aMB2Kq7vC2Yfx1ck6NB?usp=share_link)
<br>
<br>

### Programs and software


#### Programming languages

All the programming tasks in the course will be carried out with either R or on command line, although some programs we will use require other languages to be installed.

[R](https://www.r-project.org) <br>
[Java](https://www.java.com/en/download/manual.jsp) <br>
[Python](https://www.python.org/downloads/) (3.7 or higher) <br>
<br>

#### Desktop software

[R Studio](https://posit.co/download/rstudio-desktop/) (This is optional but will provide a good interface for R)<br>
[BEAST2](https://www.beast2.org) (This will install the three programs we will use in this course, BEAUti, BEAST and TreeAnnotator) <br>
[Tracer](http://tree.bio.ed.ac.uk/software/tracer/)<br>
[FigTree](http://tree.bio.ed.ac.uk/software/figtree/)<br>
Any Extended Text Viewer (e.g., [BBEdit](https://www.barebones.com/products/bbedit/) for mac, [Sublime Text](https://www.sublimetext.com) for all systems) <br>
[FASTQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)<br>
[Tablet](https://ics.hutton.ac.uk/tablet/)<br>
[AliView](https://ormbunkar.se/aliview/)<br>
<br>

#### Command line software

We will be running some programs with the Linux command line interface. 

If you are using Mac OS or Ubuntu, you can run these programs through the Terminal. If you are using Windows, please follow this guide to install a linux terminal on a Windows machine:

[Install Linux on Windows with WSL](https://learn.microsoft.com/en-us/windows/wsl/install) <br>

Also, some of these software can be easily installed using package management tools such as [homebrew]() for mac or [apt]() for Linux/Windows.

For example, with homebrew on mac, you can simply enter the following command into your terminal to install SAMtools on your system:

```bash
brew install samtools
```
Alternatively, here is a guide to install SAMtools using apt on Ubuntu:

[Install SAMtools](https://bioinformaticsreview.com/20210404/installing-samtools-on-ubuntu/) <br>


Please download the following command line software:

[Trimmomatic](https://github.com/usadellab/Trimmomatic/releases/tag/v0.39) <br>
[Burrow-Wheeler Aligner (BWA)](https://github.com/lh3/bwa) <br>
[SAMtools/BCFtools](http://www.htslib.org) <br>
[GATK](https://github.com/broadinstitute/gatk/releases) <br>
[mafft](https://mafft.cbrc.jp/alignment/software/) <br>
[IQtree](http://www.iqtree.org) <br>
[TreeTime](https://github.com/neherlab/treetime) <br>
<br>

#### R programs

These programs can be installed within R using
```R
install.packages()
```

[ape](https://cran.r-project.org/web/packages/ape/index.html) <br>
[seqinr](https://cran.r-project.org/web/packages/seqinr/index.html) <br>
[devtools](https://cran.r-project.org/web/packages/devtools/index.html)<br>
[phytools](https://cran.r-project.org/web/packages/phytools/index.html) <br>
[phylotools](https://cran.r-project.org/web/packages/phylotools/index.html) <br>
[ggplot2](https://cran.r-project.org/web/packages/ggplot2/index.html) <br>
[alluvial](https://cran.r-project.org/web/packages/ggalluvial/readme/README.html)<br>


These programs need to be installed through the 'devtools' install_github() function:

[BactDating](https://github.com/xavierdidelot/BactDating) <br>
[TreeWas](https://github.com/caitiecollins/treeWAS) <br>
<br>


### Please attempt to download all data and software. If you have any issues, please get in [contact](mailto:benjamin_sobkowiak@sfu.ca).

Click [here](README.md) to return to the course homepage.




