# Genomic Analysis and Phylodynamics Workshop 
## National Taiwan University 2025


![Title Image](Sequence_trees.png)


---


### Pre-workshop instructions

Please click the link below for instructions to download and install all data and software required for this course:

[Pre-course instructions](PreCourse_instructions.md)

---

## Schedule


| Date            | Time         | Session                                                         | Description                                                                                                                                              |
|------------------|--------------|------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------|
| **Saturday 3rd May** | 9:30–10:30   | **Lecture 1: Introduction and key concepts**                    | • Course outline<br>• Introduction to next generation sequencing and genomic epidemiology                                                               |
|                  | 10:40–12:40  | [**Practical session 1: Whole genome sequence data analysis**](Practicals/Obtaining_sequencing_data.md)    | • Obtaining sequencing data<br>• Data manipulation and QC<br>• Reference-based mapping and de novo assembly                                             |
|                  | 12:40–13:40  | **Lunch break**                                                 |                                                                                                                                                          |
|                  | 13:40–14:15  | [**Practical session 1 (cont.): Whole genome sequence data analysis**](Practicals/Obtaining_sequencing_data.md) | • Catch-up, overview, QA                                                                                                                           |
|                  | 14:15–15:15  | **Lecture 2: Variant detection and phylogenetic trees**         | • What is a variant? How do we call variants?<br>• Variant calling software and QC<br>• What are phylogenetic trees?<br>• Types of phylogenies, phylogenetic uncertainty (bootstrapping etc.) |
|                  | 15:30–18:00  | [**Practical session 2: Variant calling and maximum likelihood trees**](VariantCall.md) | • Variant calling<br>• SNP filtering and QC<br>• Building SNP matrices<br>• Aligning consensus sequences<br>• Producing ML trees                  |
| **Sunday 4th May** | 9:30–11:00   | **Lecture 3: Practical applications of WGS and phylogenetics**  | • Species identification<br>• Resistance and plasmid profiling<br>• Transmission<br>• Applications in real-world datasets                               |
|                  | 11:00–12:00  | [**Practical session 3: Timed phylogenetic trees**](Beast.md)              | • One-step timed phylogenetic tree with BEAST2 and introduction to Delphy<br>• Two-step timed phylogenies using ML + Bayesian frameworks                 |
|                  | 12:00–13:00  | **Lunch break**                                                 |                                                                                                                                                          |
|                  | 13:00–14:30  | [**Practical session 3 (cont.): Timed phylogenetic trees**](Beast.md)      | • (cont.) One-step timed phylogenetic tree <br>• Two-step timed phylogenies using ML + Bayesian frameworks                                               |
|                  | 14:45–17:00  | **Practical session 4: Transmission and profiling**            | • Identifying species, serotypes, and lineages from WGS and phylogenies<br>• Inferring transmission networks/clusters                                   |
| **Monday 5th May (Advanced)** | 9:00–10:45   | **Lecture 4: Advanced applications of WGS**                     | • Phylogeography and phylodynamics<br>• Recombination<br>• Average Nucleotide Identity (ANI)<br>• Mixed infection<br>• Fitness and selection             |
|                  | 11:00–12:00  | **Practical session 5: Mixed infection, recombination and ANI**| • Identifying mixed infection<br>• Calculating ANI<br>• Testing for recombination                                                                        |
|                  | 12:00–13:00  | **Lunch break**                                                 |                                                                                                                                                          |
|                  | 13:00–14:00  | **Practical session 5 (cont.): Mixed infection, recombination and ANI** | • (cont.) Identifying mixed infection<br>• Calculating ANI<br>• Testing for recombination                                                       |
|                  | 14:15–15:30  | **Practical session 6: Phylogeography and phylodynamics**      | • Phylogeography (ancestral state reconstruction) in R and BEAST2<br>• Phylodynamic analysis using a phylogeny (LBI, Skygrowth)                         |
| **Tuesday 6th May (Advanced)** | 9:00–12:00   | **Practical session 7: Fitness and Selection**                  | • Advanced phylodynamic analysis in BEAST2 (strain fitness, Bayesian skyline, multi-state birth-death)<br>• Site-specific selection (homoplasy, dN/dS)<br>• GWAS |
|                  | 12:00–12:30  | **Lecture 5: Closing lecture**                                  | • Course summary and feedback collection                                                                                                                 |



### View on [GitHub](https://github.com/bensobkowiak/GenomicsCourse/)

Author: Ben Sobkowiak [Email](mailto:b.sobkowiak.12@ucl.ac.uk)
