# Genomic Analysis and Phylogeography Workshop

## 5th - 9th February 2024, Simon Fraser University


![Title Image](Sequence_trees.png)

---

Please register your interest in the course [here](https://form.jotform.com/240141804355347) before the 26th January 2024

---

### Pre-workshop instructions

The following files will link you to documents with instructions for downloading relevant analysis software and visualization tools for the workshop:

Mac users: [click here](Pre-Workshop_instructions/Mac_instructions.md)<br>
Windows users: [click here](Pre-Workshop_instructions/Windows_instructions.md)<br>
Linux users: [click here](Pre-Workshop_instructions/Linux_instructions.md)<br>

### Schedule

| Time and Date                  | Topic                                           | Itinerary                                                 |
|--------------------------------|-------------------------------------------------|-----------------------------------------------------------|
| Monday 5th February <br> 10am – 12pm | [Lecture: Introduction and key concepts](Lectures/Day1_lecture.pdf) | • Course outline<br>• Genomic sequencing and data analysis<br>• Introduction to genomic epidemiology |
| Monday 5th February <br> 12:30pm – 4:30pm | [Practical session: Whole genome sequence data analysis](Practicals/WGS_Analysis.md) | • Obtaining sequencing data<br>• Raw sequencing data manipulation – Quality control and visualization<br>• Reference-based mapping |
| Tuesday 6th February <br> 9am – 10:30am | [Lecture: Variant detection and phylogenetic trees](Lectures/VariantCalling.md) | • What is a variant?<br>• How do we call variants?<br>• Variant calling software and quality control<br>• What are phylogenetic trees?<br>• Different types of phylogenies<br>• Phylogenetic uncertainty – bootstrapping etc. |
| Tuesday 6th February <br> 10:45am – 4:30pm | [Practical session: Variant calling and maximum likelihood trees](Practicals/VariantCalling_MLtrees.md) | • Variant calling from sequence alignment file<br>• SNP filtering and QC<br>• Building SNP matrices<br>• Maximum likelihood tree construction |
| Wednesday 7th February <br> 9am – 1pm | [Practical session: Phylogenetic trees](Practicals/Phylogenetics.md) | • Continue ML tree construction<br>• Timed phylogenetic tree building with BEAST2<br>• Large timed phylogenetic tree building with ML tree + BactDating |
| Thursday 8th February <br> 9am – 10:30am | [Lecture: Phylogeography and phylodynamics](Lectures/Phylogeography_Phylodynamics.md) | • What are phylogeography and phylodynamics?<br>• Analysis tools and software<br>• Applications to real-world datasets – what can we learn |
| Thursday 8th February <br> 10:45am – 4:30pm | [Practical session: Epidemiological analysis using phylogenetic trees](Practicals/Phylodynamics.md) | • Phylodynamic analysis in BEAST2<br>• Ancestral state reconstruction<br>• Selection coefficients<br>• Inference of population dynamics |
| Friday 9th February <br> 9am – 1pm  | [Practical session: Self guided practical: Full pipeline](Practicals/Full_pipeline_test.md) | • Independent application of the full analysis pipeline to a novel dataset<br>• From raw sequence data to phylogeographic analysis |


Author: Dr. Ben Sobkowiak (b.sobkowiak.12 at ucl.ac.uk)
